smoke-test/urls.test.js

```diff
@@ -0,0 +1,52 @@
+const {readFileSync} = require('fs');
+const testCases = [];
+
+// Load a suite of smoke tests from a simple text file. File is of the format:
+// [path], [comma separated text assertions]
+const text = readFileSync(require.resolve('./urls.txt'), 'utf8').trim();
+for (const line of text.split(/\r?\n/)) {
+  if (line.length === 0) continue; // allow for blank lines.
+  if (line.match(/^#/)) continue; // allow for comments.
+  const testCase = {assertions: []};
+  line.split(/,/).forEach((token, i) => {
+    if (i === 0) {
+      // The first token is the path that we wish to traverse the headless
+      // browser to:
+      testCase.path = token;
+    } else {
+      // The remaining tokens represent blocks of text in the rendered page
+      // we should check for:
+      testCase.assertions.push(token.trimLeft());
+    }
+  });
+  testCases.push(testCase);
+}
+
+describe("smoke test rendering of paths in urls.txt", () => {
+  for (const testCase of testCases) {
+    // eslint-disable-next-line no-loop-func
+    it(`it appropriately renders ${testCase.path}`, async () => {
+      const resp = await page.goto(`${BASE_URL}${testCase.path}`, { waitUntil: 'networkidle2' });
+      expect(resp.status()).toEqual(200);
+      // Create a Devtools session for fetching MIME HTML snapshot:
+      const client = await page.target().createCDPSession();
```

Comment:
(A note, not a request for this PR) Would be worth investigating [the runtime API](https://chromedevtools.github.io/devtools-protocol/tot/Runtime/#event-consoleAPICalled) to check for any `console.error()` calls during each test and failing on these.

Reply:
this might be interesting to try for sure, I can experiment.

Reply:
Let's not worry about this here, was more a note for the future. I'm sure it will highlight a scary number of console errors!

Reply:
I wonder if we could consider doing this in a follow up? don't want this to get too complicated.